test/google_scraping_web/api/v1/upload_keyword_controller_test.exs

```diff
@@ -0,0 +1,123 @@
+defmodule GoogleScrapingWeb.Api.V1.UploadKeywordControllerTest do
+  use GoogleScrapingWeb.ConnCase, async: true
+
+  describe "post create/2" do
+    test "given the valid keywords file, returns 201 status with success message", %{conn: conn} do
+      use_cassette "google/valid_file" do
+        user = insert(:user)
+        upload_file = keyword_file_fixture("valid.csv")
+
+        conn =
+          conn
+          |> token_auth_user(user)
+          |> post(Routes.api_upload_keyword_path(conn, :create), %{
+            keyword_csv_file: upload_file
+          })
+
+        assert %{
+                 "data" => %{
+                   "attributes" => %{
+                     "message" => "Keywords were uploaded!"
+                   },
+                   "id" => _,
+                   "relationships" => %{},
+                   "type" => "upload_keyword"
+                 },
+                 "included" => []
+               } = json_response(conn, 201)
+      end
+    end
+  end
+
+  test "given an empty file, returns 422 status with error details", %{conn: conn} do
+    user = insert(:user)
+    upload_file = keyword_file_fixture("empty.csv")
+
+    conn =
+      conn
+      |> token_auth_user(user)
+      |> post(Routes.api_upload_keyword_path(conn, :create), %{
+        keyword_csv_file: upload_file
+      })
+
+    assert %{"errors" => [%{"detail" => "The file is empty", "status" => "unprocessable_entity"}]} ==
```

Comment:
We keep the matching value at the right side when we assert with `==` to make it more redable

so it should be `assert json_response(conn, 422) ==  %{"errors" ...`

Reply:
Yes, thanks. I always forgot about this one.